FAQ.md

```diff
@@ -0,0 +1,80 @@
+
+- `dopri5`<br>
+	The `dopri5` ODE solver stores at least 6 evaluations of the ODE, then takes a step using a linear combination of them. The diagram below illustrates it: the evaluations marked with `o` are on the estimated path, the others with `x` are not. The first two are for selecting the initial step size.
+
+	0  1 |  2  3  4  5  6  7 |  8  9  10 12 13 14
+	o  x |  x  x  x  x  x  o |  x  x  x  x  x  o
+
+
+**How do I obtain evaluations on the estimated path when using an adaptive solver ?**<br>
+The argument `t` of `odeint` specifies what times should the ODE solver output.
+odeint(func, x0, t=torch.linspace(0, 1, 50))
+
+Note that the ODE solver will always integrate from `min t(0)` to `max t(1)`, and the intermediate values of `t` have no effect on how the ODE the solved. Computing intermediate values of t have a very small cost.
+
+**What non-linearities should I use in my Neural ODE?**<br>
+Avoid non-smooth non-linearities such as ReLU and LeakyReLU.<br>
+Prefer non-linearities with a theoretically unique adjoint/gradient such as Softplus.
+
+**Where is backpropagation for the Neural ODE defined?**<br>
+It's defined [here](https://github.com/rtqichen/torchdiffeq/blob/master/torchdiffeq/_impl/adjoint.py#L105) if you use the adjoing method `odeint_adjoint`.
+
+**What are Tableaus?**<br>
+Tableaus are ways to describe coefficients for [RK methods](https://en.wikipedia.org/wiki/Runge%E2%80%93Kutta_methods). The particular set of coefficients used on this repo was taken from [here](https://www.ams.org/journals/mcom/1986-46-173/S0025-5718-1986-0815836-3/).
+
+**How do I install the repo on Windows?**<br>
+try downloading the code directly and just running python setup.py install.
+https://stackoverflow.com/questions/52528955/installing-a-python-module-from-github-in-windows-10
+
+**What is the most memory-expensive operation during training?**<br>
+The most memory-expensive operation is the single [backward call](https://github.com/rtqichen/torchdiffeq/blob/master/torchdiffeq/_impl/adjoint.py#L41) made to the network.
+    
+**My Neural ODE's numerical solution is farther away from the target than the initial value**<br>
+Most tricks for initializing residual nets (like zeroing the weights of the last layer) should help for ODEs as well. This will initialize the ODE as an identity.
+
+
+**My Neural ODE takes too long to train**<br>
+This might be because you're running on CPU. Being extremely slow on CPU is expected, as training requires evaluating a neural net multiple times.
+
+
+**My Neural ODE produces underflow in dt when using dopri5**<br>
+This is a problem of the ODE becoming stiff, essentially acting too erratic in a region and the step size becomes so close to zero that no progress can be made in the solver. We were able to avoid this with regularization such as weight decay and using "nice" activation functions, but YMMV. Another option is to use a fixed step size solver to get an approximate solution.
```

Comment:
Let's remove the last sentence "Another option is to use a fixed step size solver to get an approximate solution" because the resulting model would just be a resnet and won't be a proper ODE anymore.